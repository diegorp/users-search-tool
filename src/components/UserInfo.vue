<template>
    <div class="user">
        <div class="user__info">
            <p class="user__info__name">{{ user.name }}</p>
            <p class="user__info__email">{{ user.email }}</p>
            <p class="user__info__phone">{{ user.phone }}</p>
        </div>
        <div class="user__company">
            <p class="user__company__name">{{ user.company.name }}</p>
            <p class="user__company__phrase">{{ user.company.catchPhrase }}</p>
        </div>
    </div>
</template>

<script>
export default {
    name: 'UserInfo',
    props: {
        user: Object
    },
}
</script>

<style scoped lang="scss">
    .user {
        display: flex;
        justify-content: space-between;
        background-color: #f6f6f6;

        &__info {
            padding: 0 6px;
            text-align: left;
            flex: 50%;
    
            &__email,
            &__phone {
                font-size: 0.75em;
            }
        }

        &__company {
            padding: 0 6px;
            flex: 50%;
            
            &__phrase {
                font-size: 0.75em;
            }
        }
    }
</style>