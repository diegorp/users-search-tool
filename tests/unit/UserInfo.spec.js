import { shallowMount } from "@vue/test-utils";
import UserInfo from '@/components/UserInfo.vue'

describe('UserInfo.vue', () => {
    let wrapper;
    const user = {
        id: 9,
        name: 'John Doe',
        email: 'johndoe@mail.com',
        phone: '(775)976-6794 x41206',
        company: {
            name: 'Alkimii',
            catchPhrase: 'Integrated Applications made by our hotel people for your hotel people.'
        }
    }

    beforeEach(() => {
        wrapper = shallowMount(UserInfo, {
            props: { user }
        });
    });

    it('should render the component', () => {
        expect(wrapper).toMatchSnapshot();
    });
});