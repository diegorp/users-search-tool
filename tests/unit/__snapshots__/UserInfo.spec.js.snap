// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo.vue should render the component 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="user"
      >
        <div
          class="user__info"
        >
          <p
            class="user__info__name"
          >
            John Doe
          </p>
          <p
            class="user__info__email"
          >
            <img
              src=""
            />
            johndoe@mail.com
          </p>
          <p
            class="user__info__phone"
          >
            <img
              src=""
            />
            (775)976-6794 x41206
          </p>
        </div>
        <div
          class="user__company"
        >
          <p
            class="user__company__name"
          >
            Alkimii
          </p>
          <p
            class="user__company__phrase"
          >
            Integrated Applications made by our hotel people for your hotel people.
          </p>
        </div>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {},
      "da": null,
      "data": Object {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {},
      "refs": Object {
        "VTU_COMPONENT": Object {
          "hasOwnProperty": [Function],
          "user": Object {
            "company": Object {
              "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
              "name": "Alkimii",
            },
            "email": "johndoe@mail.com",
            "id": 9,
            "name": "John Doe",
            "phone": "(775)976-6794 x41206",
          },
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": Array [],
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": Object {},
      "slots": Object {},
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "constructor": 0,
            "hasOwnProperty": 4,
            "isVueInstance": 0,
            "toJSON": 0,
            "user": 3,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "hasOwnProperty": [Function],
            "user": Object {
              "company": Object {
                "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                "name": "Alkimii",
              },
              "email": "johndoe@mail.com",
              "id": 9,
              "name": "John Doe",
              "phone": "(775)976-6794 x41206",
            },
          },
          "da": null,
          "data": Object {},
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {
            "user": Object {
              "company": Object {
                "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                "name": "Alkimii",
              },
              "email": "johndoe@mail.com",
              "id": 9,
              "name": "John Doe",
              "phone": "(775)976-6794 x41206",
            },
          },
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {
              "user": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
            },
            Array [],
          ],
          "provides": Object {},
          "proxy": Object {
            "hasOwnProperty": [Function],
            "user": Object {
              "company": Object {
                "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                "name": "Alkimii",
              },
              "email": "johndoe@mail.com",
              "id": 9,
              "name": "John Doe",
              "phone": "(775)976-6794 x41206",
            },
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": Array [],
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "John Doe",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <p
                      class="user__info__name"
                    >
                      John Doe
                    </p>,
                    "key": null,
                    "patchFlag": 1,
                    "props": Object {
                      "class": "user__info__name",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "p",
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": Array [
                      Object {
                        "__v_isVNode": true,
                        "__v_skip": true,
                        "anchor": null,
                        "appContext": null,
                        "children": null,
                        "component": null,
                        "dirs": null,
                        "dynamicChildren": null,
                        "dynamicProps": null,
                        "el": <img
                          src=""
                        />,
                        "key": null,
                        "patchFlag": -1,
                        "props": Object {
                          "src": "",
                        },
                        "ref": null,
                        "scopeId": null,
                        "shapeFlag": 1,
                        "slotScopeIds": null,
                        "ssContent": null,
                        "ssFallback": null,
                        "staticCount": 0,
                        "suspense": null,
                        "target": null,
                        "targetAnchor": null,
                        "transition": null,
                        "type": "img",
                      },
                      Object {
                        "__v_isVNode": true,
                        "__v_skip": true,
                        "anchor": null,
                        "appContext": null,
                        "children": "johndoe@mail.com",
                        "component": null,
                        "dirs": null,
                        "dynamicChildren": null,
                        "dynamicProps": null,
                        "el": johndoe@mail.com,
                        "key": null,
                        "patchFlag": 1,
                        "props": null,
                        "ref": null,
                        "scopeId": null,
                        "shapeFlag": 8,
                        "slotScopeIds": null,
                        "ssContent": null,
                        "ssFallback": null,
                        "staticCount": 0,
                        "suspense": null,
                        "target": null,
                        "targetAnchor": null,
                        "transition": null,
                        "type": Symbol(Text),
                      },
                    ],
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <p
                      class="user__info__email"
                    >
                      <img
                        src=""
                      />
                      johndoe@mail.com
                    </p>,
                    "key": null,
                    "patchFlag": 0,
                    "props": Object {
                      "class": "user__info__email",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 17,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "p",
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": Array [
                      Object {
                        "__v_isVNode": true,
                        "__v_skip": true,
                        "anchor": null,
                        "appContext": null,
                        "children": null,
                        "component": null,
                        "dirs": null,
                        "dynamicChildren": null,
                        "dynamicProps": null,
                        "el": <img
                          src=""
                        />,
                        "key": null,
                        "patchFlag": -1,
                        "props": Object {
                          "src": "",
                        },
                        "ref": null,
                        "scopeId": null,
                        "shapeFlag": 1,
                        "slotScopeIds": null,
                        "ssContent": null,
                        "ssFallback": null,
                        "staticCount": 0,
                        "suspense": null,
                        "target": null,
                        "targetAnchor": null,
                        "transition": null,
                        "type": "img",
                      },
                      Object {
                        "__v_isVNode": true,
                        "__v_skip": true,
                        "anchor": null,
                        "appContext": null,
                        "children": "(775)976-6794 x41206",
                        "component": null,
                        "dirs": null,
                        "dynamicChildren": null,
                        "dynamicProps": null,
                        "el": (775)976-6794 x41206,
                        "key": null,
                        "patchFlag": 1,
                        "props": null,
                        "ref": null,
                        "scopeId": null,
                        "shapeFlag": 8,
                        "slotScopeIds": null,
                        "ssContent": null,
                        "ssFallback": null,
                        "staticCount": 0,
                        "suspense": null,
                        "target": null,
                        "targetAnchor": null,
                        "transition": null,
                        "type": Symbol(Text),
                      },
                    ],
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <p
                      class="user__info__phone"
                    >
                      <img
                        src=""
                      />
                      (775)976-6794 x41206
                    </p>,
                    "key": null,
                    "patchFlag": 0,
                    "props": Object {
                      "class": "user__info__phone",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 17,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "p",
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="user__info"
                >
                  <p
                    class="user__info__name"
                  >
                    John Doe
                  </p>
                  <p
                    class="user__info__email"
                  >
                    <img
                      src=""
                    />
                    johndoe@mail.com
                  </p>
                  <p
                    class="user__info__phone"
                  >
                    <img
                      src=""
                    />
                    (775)976-6794 x41206
                  </p>
                </div>,
                "key": null,
                "patchFlag": 0,
                "props": Object {
                  "class": "user__info",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Alkimii",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <p
                      class="user__company__name"
                    >
                      Alkimii
                    </p>,
                    "key": null,
                    "patchFlag": 1,
                    "props": Object {
                      "class": "user__company__name",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "p",
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Integrated Applications made by our hotel people for your hotel people.",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <p
                      class="user__company__phrase"
                    >
                      Integrated Applications made by our hotel people for your hotel people.
                    </p>,
                    "key": null,
                    "patchFlag": 1,
                    "props": Object {
                      "class": "user__company__phrase",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "p",
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="user__company"
                >
                  <p
                    class="user__company__name"
                  >
                    Alkimii
                  </p>
                  <p
                    class="user__company__phrase"
                  >
                    Integrated Applications made by our hotel people for your hotel people.
                  </p>
                </div>,
                "key": null,
                "patchFlag": 0,
                "props": Object {
                  "class": "user__company",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "John Doe",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <p
                  class="user__info__name"
                >
                  John Doe
                </p>,
                "key": null,
                "patchFlag": 1,
                "props": Object {
                  "class": "user__info__name",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "p",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "johndoe@mail.com",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": johndoe@mail.com,
                "key": null,
                "patchFlag": 1,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 8,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Text),
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "(775)976-6794 x41206",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": (775)976-6794 x41206,
                "key": null,
                "patchFlag": 1,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 8,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Text),
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Alkimii",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <p
                  class="user__company__name"
                >
                  Alkimii
                </p>,
                "key": null,
                "patchFlag": 1,
                "props": Object {
                  "class": "user__company__name",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "p",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Integrated Applications made by our hotel people for your hotel people.",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <p
                  class="user__company__phrase"
                >
                  Integrated Applications made by our hotel people for your hotel people.
                </p>,
                "key": null,
                "patchFlag": 1,
                "props": Object {
                  "class": "user__company__phrase",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "p",
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="user"
            >
              <div
                class="user__info"
              >
                <p
                  class="user__info__name"
                >
                  John Doe
                </p>
                <p
                  class="user__info__email"
                >
                  <img
                    src=""
                  />
                  johndoe@mail.com
                </p>
                <p
                  class="user__info__phone"
                >
                  <img
                    src=""
                  />
                  (775)976-6794 x41206
                </p>
              </div>
              <div
                class="user__company"
              >
                <p
                  class="user__company__name"
                >
                  Alkimii
                </p>
                <p
                  class="user__company__phrase"
                >
                  Integrated Applications made by our hotel people for your hotel people.
                </p>
              </div>
            </div>,
            "key": null,
            "patchFlag": 0,
            "props": Object {
              "class": "user",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "name": "UserInfo",
            "props": Object {
              "user": [Function],
            },
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="user"
        >
          <div
            class="user__info"
          >
            <p
              class="user__info__name"
            >
              John Doe
            </p>
            <p
              class="user__info__email"
            >
              <img
                src=""
              />
              johndoe@mail.com
            </p>
            <p
              class="user__info__phone"
            >
              <img
                src=""
              />
              (775)976-6794 x41206
            </p>
          </div>
          <div
            class="user__company"
          >
            <p
              class="user__company__name"
            >
              Alkimii
            </p>
            <p
              class="user__company__phrase"
            >
              Integrated Applications made by our hotel people for your hotel people.
            </p>
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
          "user": Object {
            "company": Object {
              "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
              "name": "Alkimii",
            },
            "email": "johndoe@mail.com",
            "id": 9,
            "name": "John Doe",
            "phone": "(775)976-6794 x41206",
          },
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "name": "UserInfo",
          "props": Object {
            "user": [Function],
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="user"
        >
          <div
            class="user__info"
          >
            <p
              class="user__info__name"
            >
              John Doe
            </p>
            <p
              class="user__info__email"
            >
              <img
                src=""
              />
              johndoe@mail.com
            </p>
            <p
              class="user__info__phone"
            >
              <img
                src=""
              />
              (775)976-6794 x41206
            </p>
          </div>
          <div
            class="user__company"
          >
            <p
              class="user__company__name"
            >
              Alkimii
            </p>
            <p
              class="user__company__phrase"
            >
              Integrated Applications made by our hotel people for your hotel people.
            </p>
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.33",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "hasOwnProperty": [Function],
    "user": Object {
      "company": Object {
        "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
        "name": "Alkimii",
      },
      "email": "johndoe@mail.com",
      "id": 9,
      "name": "John Doe",
      "phone": "(775)976-6794 x41206",
    },
  },
  "isDisabled": [Function],
  "rootVM": Object {},
  "wrapperElement": <div
    class="user"
  >
    <div
      class="user__info"
    >
      <p
        class="user__info__name"
      >
        John Doe
      </p>
      <p
        class="user__info__email"
      >
        <img
          src=""
        />
        johndoe@mail.com
      </p>
      <p
        class="user__info__phone"
      >
        <img
          src=""
        />
        (775)976-6794 x41206
      </p>
    </div>
    <div
      class="user__company"
    >
      <p
        class="user__company__name"
      >
        Alkimii
      </p>
      <p
        class="user__company__phrase"
      >
        Integrated Applications made by our hotel people for your hotel people.
      </p>
    </div>
  </div>,
}
`;
